File: app/src/main/java/com/jwu5/giphyapp/model/GiphyModel.java
package com.jwu5.giphyapp.model;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

/**
 * Created by Jiawei on 8/17/2017.
 */
public class GiphyModel implements Serializable {
    public static final String PREVIEW_URL = "https://i.giphy.com/media/";
    @SerializedName("type")
    @Expose
    private String type;
    @SerializedName("id")
    @Expose
    private String id;
    @SerializedName("url")
    @Expose
    private String url;

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getUrl() {
        return PREVIEW_URL + id + "/200w.gif";
    }

    public void setUrl(String url) {
        this.url = url;
    }
}

File: app/src/main/java/com/jwu5/giphyapp/network/model/GiphyModel.kt
package com.jwu5.giphyapp.network.model

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

import java.io.Serializable

/**
 * Created by Jiawei on 8/17/2017.
 */
class GiphyModel {
    @SerializedName("type")
    @Expose
    var type: String? = null
    @SerializedName("id")
    @Expose
    var id: String? = null
    @SerializedName("url")
    @Expose
    var url: String? = null
        get() = PREVIEW_URL + id + "/200w.gif"

    companion object {
        val PREVIEW_URL = "https://i.giphy.com/media/"
    }
}
